src/main/java/net/fabricmc/loader/minecraft/McVersionLookup.java

```diff
@@ -290,7 +293,20 @@ private static String normalizeVersion(String name, String release) {
 				matcher = PRE_RELEASE_PATTERN.matcher(name);
 
 				if (matcher.matches()) {
-					name = String.format("rc.%s", matcher.group(1));
+					boolean legacyVersion;
+
+					try {
+						legacyVersion = SemanticVersionPredicateParser.create("<=1.16").test(new SemanticVersionImpl(release, false));
+					} catch (VersionParsingException e) {
+						throw new RuntimeException("Failed to parse version");
+					}
+
+					// Mark pre-releases as beta expect for legacy versions where these were previously marked as rc
```

Comment:
except

Reply:
The entire comment could be clarified a bit:

```suggestion
					// Mark pre-releases as beta expect for legacy versions where these were previously marked as rc
					// Mark pre-releases as 'beta' versions, except for version 1.16 and before, where they are 'rc'
```